geth/account/importer_test.go

```diff
@@ -0,0 +1,81 @@
+package account
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/golang/mock/gomock"
+	"github.com/status-im/status-go/extkeys"
+	"github.com/stretchr/testify/require"
+)
+
+func TestManager_ImportExtendedKey_Success(t *testing.T) {
+	extKey := &extkeys.ExtendedKey{}
+	password := "123"
+	acc := accounts.Account{Address: ethcommon.Address{}}
+	key := keystore.Key{
+		PrivateKey: &ecdsa.PrivateKey{
+			PublicKey: ecdsa.PublicKey{},
+		},
+	}
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	keystoreMock := NewMockaccountKeyStorer(ctrl)
+	keystoreMock.EXPECT().ImportExtendedKey(extKey, password).
+		Times(1).Return(acc, nil)
+
+	keystoreMock.EXPECT().AccountDecryptedKey(acc, password).
+		Times(1).Return(acc, &key, nil)
+
+	e := extendedKeyImporterBase{}
+	addr, pub, err := e.Import(keystoreMock, extKey, password)
```

Comment:
Not yet sure about this mock generator, have to dig into it. Who is responsible for the returned `addr` and `pub`. If I follow the code above the `addr` is returned by the `keyStore`. And in our case here that's the mock. So the test below for `addr` tests the returned value of the mock, doesn't it?

Reply:
@themue Yes, it's true. ```addr``` check is redundant.